src/GeneratesUuid.php

```diff
@@ -61,7 +61,7 @@ public static function bootGeneratesUuid()
                 $uuid = $uuid->fromString(strtolower($model->attributes[$model->uuidColumn()]));
             }
 
-            $model->attributes[$model->uuidColumn()] = $model->hasCast('uuid') ? $uuid->getBytes() : $uuid->toString();
+            $model->attributes[$model->uuidColumn()] = $model->hasCast($model->uuidColumn()) ? $uuid->getBytes() : $uuid->toString();
```

Comment:
Just a user passing by, is it better to add type checking as well? i.e.
`$model->hasCast($model->uuidColumn(), 'uuid')`